### PR TITLE
Add `predict_k` method in CPT

### PR DESCRIPTION
**Pull request by bluesheeptoken**

### CODE REVIEW

File: cpt/cpt.pyx [hunk 1]
```diff
@@ -141,30 +141,92 @@ cdef class Cpt:
...
+        --------
+
+        >>> model = Cpt()
+
+        >>> model.fit([['hello', 'world'],
+             ['hello', 'this', 'is', 'me'],
+             ['hello', 'me']
+            ])
+
+        >>> model.predict([['hello'], ['hello', 'this']])
```
Comment (kimci86) [inline cpt/cpt.pyx:191]: This should be `predict_k`

File: cpt/cpp_sources/Scorer.cpp [hunk 2]
```diff
@@ -30,3 +31,28 @@ int Scorer::get_best_prediction() const
 {
     return std::distance(m_data.begin(), std::max_element(m_data.begin(), m_data.end()));
 }
+
+std::vector<int> Scorer::get_best_k_predictions(std::size_t k) const
```
Comment (kimci86) [inline cpt/cpp_sources/Scorer.cpp:35]: Good but this should use `std::partial_sort_copy`.

File: cpt/cpp_sources/Scorer.cpp [hunk 3]
```diff
@@ -30,3 +31,28 @@ int Scorer::get_best_prediction() const
...
+{
+    std::priority_queue<std::pair<double, int>, std::vector<std::pair<double, int>>, std::greater<std::pair<double, int>>> q;
```
Comment (kimci86) [inline cpt/cpp_sources/Scorer.cpp:37]: Include `<functional>` to use `std::greater`

File: cpt/cpp_sources/Scorer.cpp (hunk 2 above)
Comment (bluesheeptoken) [reply cpt/cpp_sources/Scorer.cpp:35]: I should probably at least do some benchmarks. I am just afraid that the full copy of the `m_data` costs a bit too much.

I will do benchmarks, and depending on them I will choose (I prefer `partial_sort_copy` for readability)

**Comment by kimci86** [comment]
I rewrote the `get_best_k_predictions`. It is more readable, but I am not sure about performance. Feel free to revert my commit.

**Comment by bluesheeptoken** [comment]
FYI, with k=10:

priority queue:
```
 Performance counter stats for 'python3 benchmark/benchmark.py':

         13 613,46 msec task-clock                #    1,000 CPUs utilized          
                38      context-switches          #    0,003 K/sec                  
                 2      cpu-migrations            #    0,000 K/sec                  
            43 010      page-faults               #    0,003 M/sec                  
    40 704 692 056      cycles                    #    2,990 GHz                    
   102 777 331 787      instructions              #    2,52  insn per cycle         
    21 136 261 338      branches                  # 1552,600 M/sec                  
        69 800 313      branch-misses             #    0,33% of all branches        

      13,615035167 seconds time elapsed

      13,526541000 seconds user
       0,087990000 seconds sys
```
`partial_sort_copy`:
```       
 Performance counter stats for 'python3 benchmark/benchmark.py':

         13 649,25 msec task-clock                #    1,000 CPUs utilized          
                16      context-switches          #    0,001 K/sec                  
                 0      cpu-migrations            #    0,000 K/sec                  
            43 014      page-faults               #    0,003 M/sec                  
    40 861 383 553      cycles                    #    2,994 GHz                    
   102 033 828 743      instructions              #    2,50  insn per cycle         
    21 058 381 814      branches                  # 1542,823 M/sec                  
        68 191 679      branch-misses             #    0,32% of all branches        

      13,649734480 seconds time elapsed

      13,577593000 seconds user
       0,072008000 seconds sys
```

Which are quite similar, I will keep `partial_sort_copy` for readability.

**Review by kimci86 — CHANGES_REQUESTED** [review]
Fix `test_predict_k`.
Otherwise looks good to me.

**Review by kimci86 — APPROVED** [review]

**Pull request merged**